src/Akka.Streams.Kafka/Stages/Consumers/Concrete/PlainSubSourceStage.cs

```diff
@@ -50,19 +49,46 @@ public PlainSubSourceStage(ConsumerSettings<K, V> settings, IAutoSubscription su
             Subscription = subscription;
             GetOffsetsOnAssign = getOffsetsOnAssign;
             OnRevoke = onRevoke;
+            _subSourceStageLogicFactory = new PlainSubSourceStageLogicFactory();
         }
 
-        /// <inheritdoc />
+        private class
+            PlainSubSourceStageLogicFactory : SubSourceLogic.ISubSourceStageLogicFactory<K, V, ConsumeResult<K, V>>
```

Comment:
Same idea as before - move all of the stream stage construction inside its own factory.